acceptance/suites/pre_suite/puppet3_compat/70_install_puppet.rb

```diff
@@ -9,5 +9,13 @@
 
 install_puppet_on(nonmaster_agents, {:version => puppet_version})
 
+variant = master['platform'].variant
```

Comment:
This is a lot better. I've updated the code, lemme know if you like the new approach.